tests/Feature/TagControllerTest.php

```diff
@@ -0,0 +1,171 @@
+
+        $response->assertJson(['message' => 'No hi ha tags a la base de dades']);
+    }
+
+    public function testStoreTag()
+    {
+        $tagData = Tag::factory()->make()->toArray();
+
+        $response = $this->json('POST', '/api/v1/tags', $tagData);
+
+        $response->assertStatus(201);
+
+        $response->assertJsonStructure([
+            'data' => [
+                'id',
+                'tag_name',
+            ],
+            'message',
+        ]);
+    }
+
+    public function testStoreFailsWhenTagNameIsMissing()
+    {
+        $response = $this->json('POST', '/api/v1/tags', []);
+
+        $response->assertStatus(422);
+
+        $response->assertJsonStructure([
+            'message',
+            'errors' => [
+                'tag_name'
+            ]
+        ]);
+
+        $response->assertJsonFragment([
+            'tag_name' => ['El camp tag name és obligatori.'],
+        ]);
+    }
+
+    public function testShowReturnsTag()
+    {
+        $tag = Tag::factory()->create();
+
+        $response = $this->json('GET', "/api/v1/tags/{$tag->id}");
+
+        $response->assertStatus(200);
+
+        $response->assertJsonStructure([
+            'data' => [
+                'id',
+                'tag_name',
+            ],
+        ]);
+
+        // Assert the response data matches the created tag
+        $response->assertJson([
+            'data' => [
+                'id' => $tag->id,
+                'tag_name' => $tag->tag_name,
+            ],
+        ]);
+    }
+
+    public function testUpdateTagSuccessfully()
+    {
+        $tag = Tag::factory()->create();
+
+        $updatedData = [
+            'tag_name' => 'Updated Tag Name',
+        ];
+
+        $response = $this->json('PUT', "/api/v1/tags/{$tag->id}", $updatedData);
+
+        $response->assertStatus(200);
+
+        $response->assertJsonStructure([
+            'data' => [
+                'id',
+                'tag_name',
+            ],
+            'message',
+        ]);
+
+        $response->assertJson([
+            'data' => [
+                'id' => $tag->id,
+                'tag_name' => $updatedData['tag_name'],
+            ],
+            'message' => __('Tag updated successfully'),
+        ]);
+
+        // Refresh the tag from the database
+        $tag->refresh();
+
+        // Assert the tag in the database has been updated
+        $this->assertEquals($updatedData['tag_name'], $tag->tag_name);
+    }
+
+    public function testDestroyTagSuccessfully()
+    {
+        $tag = Tag::factory()->create();
+
+        $response = $this->json('DELETE', "/api/v1/tags/{$tag->id}");
+
+        $response->assertStatus(200);
+
+        $response->assertJson(['message' => __('Tag deleted successfully')]);
+
+        // Attempt to retrieve the deleted tag from the database
+        $retrievedTag = Tag::find($tag->id);
+
+        // Assert that the tag has been deleted
+        $this->assertNull($retrievedTag);
```

Comment:
We also can test if the id is not at database.
```suggestion
        $this->assertNull($retrievedTag);
        $this->assertDatabaseMissing('tags', ['id' => $tag->id]);
```

Reply:
tnxs! I'll sum this line to the test!